inference-engine/src/mkldnn_plugin/mkldnn_infer_request.cpp

```diff
@@ -425,10 +425,10 @@ void MKLDNNPlugin::MKLDNNInferRequest::SetBlob(const std::string& name, const In
                     IE_THROW(ParameterMismatch) << "Failed to set input blob. Blocking descriptor mismatch.";
             }
 
-            const auto &actualDesc = graph->getInputNodeByName(name)->getChildEdgesAtPort(0)[0]->getMemory().getDesc();
+            const auto actualDesc = MemoryDescUtils::convertToBlockedMemoryDesc(graph->getInputNodeByName(name)->getBaseMemDescAtOutputPort(0));
             if (blobDesc.getLayout() != InferenceEngine::Layout::ANY &&
-                actualDesc.isCompatible(MemoryDescUtils::convertToCpuBlockedMemoryDesc(blobDesc)) &&
-                graph->_normalizePreprocMap.find(name) == graph->_normalizePreprocMap.end() && !graph->getProperty().batchLimit) {
+                actualDesc->cloneWithNewDims(blobDesc.getDims())->isCompatible(MemoryDescUtils::convertToCpuBlockedMemoryDesc(blobDesc)) &&
```

Comment:
We definitely need detailed comment that describes the behavior.

Reply:
comment added